Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CentralRepoDataValidator.java

```diff
@@ -0,0 +1,150 @@
+        }
+    }
+    
+    /**
+     * Validate the data.  Converts text to lower case, and ensures that the
+     * data is a valid string of the format expected given the attributeType.
+     *  
+     * @param attributeTypeId correlation type of data
+     * @param data data to validate
+     * 
+     * @return normalized data
+     */
+    public static String validate(int attributeTypeId, String data) throws CentralRepoValidationException {
+        try {
+            List<CorrelationAttribute.Type> defaultTypes = CorrelationAttribute.getDefaultCorrelationTypes();
+            Optional<CorrelationAttribute.Type> typeOption = defaultTypes.stream().filter(attributeType -> attributeType.getId() == attributeTypeId).findAny();
+            
+            if(typeOption.isPresent()){
+                CorrelationAttribute.Type type = typeOption.get();
+                return CentralRepoDataValidator.validate(type, data);
+            } else {
+                throw new CentralRepoValidationException(String.format("Given attributeTypeId did not correspond to any known Attribute: %s", attributeTypeId));
+            }
+        } catch (EamDbException ex) {
+            throw new CentralRepoValidationException(ex);
+        }
+    }
+
+    private static String validateMd5(String data) throws CentralRepoValidationException {
+        final String errorMessage = "Data purporting to be an MD5 was found not to comform to expected format: %s";
+        if(data == null){
+            throw new CentralRepoValidationException(String.format(errorMessage, data));
+        }
+        final String validMd5Regex = "^[a-fA-F0-9]{32}$";
+        final String dataLowered = data.toLowerCase();
+        if(dataLowered.matches(validMd5Regex)){
+            return dataLowered;
+        } else {
+            throw new CentralRepoValidationException(String.format(errorMessage, data));
+        }
+    }
+
+    private static String validateDomain(String data) throws CentralRepoValidationException {
+        DomainValidator validator = DomainValidator.getInstance(true);
+        if(validator.isValid(data)){
+            return data.toLowerCase();
+        } else {
+            throw new CentralRepoValidationException(String.format("Data was expected to be a valid domain: %s", data));
+        }
+    }
+
+    private static String validateEmail(String data) throws CentralRepoValidationException {
+        EmailValidator validator = EmailValidator.getInstance(true, true);
+        if(validator.isValid(data)){
+            return data.toLowerCase();
+        } else {
+            throw new CentralRepoValidationException(String.format("Data was expected to be a valid email address: %s", data));
+        }
+    }
+
+    @SuppressWarnings("DeadBranch")
+    private static String validatePhone(String data) throws CentralRepoValidationException {
+        //TODO implement for real and get rid of suppression
+        if(true){
```

Comment:
For the current story in JIRA, although we are holding off on validation using a library, there is still a validation and normalization step to be done:

"PHONE: Verify it is only numbers and '+'. Strip spaces, dashes, and parentheses. As a longer-term step we'll validate using a Google library, but let's keep that out of this story (see JIRA-3869)."

Please resolve the TODO as described, for now, thanks!